extensions/amp-auto-ads/0.1/anchor-ad-strategy.js

```diff
@@ -38,31 +39,24 @@ export class AnchorAdStrategy {
 
     /** @const @private {!JSONType} */
     this.configObj_ = configObj;
+
+    /** @const @private {JsonObject<string, string>} */
+    this.anchorAdAttributes_ = anchorAdAttributes;
   }
 
   /**
    * @return {!Promise<boolean>} Resolves when the strategy is complete.
    */
   run() {
-    const filledAnchorEnabled = this.isFilledAnchorAdEnabled_();
-    const noFillAnchorEnabled = this.isNoFillAnchorAdEnabled_();
-
     if (this.hasExistingStickyAd_()) {
-      if (noFillAnchorEnabled) {
-        user().warn(
-          TAG,
-          'Auto ads may not work because of already existing <amp-sticky-ad>.'
-        );
-      } else {
-        user().warn(
-          TAG,
-          'Auto Ads is unable to create an anchor ad because of already existing <amp-sticky-ad>.'
-        );
-      }
+      user().warn(
+        TAG,
+        'Auto ads may not work because of already existing <amp-sticky-ad>.'
+      );
       return Promise.resolve(false);
     }
 
-    if (!filledAnchorEnabled && !noFillAnchorEnabled) {
+    if (!this.isFilledAnchorAdEnabled_() && !this.anchorAdAttributes_) {
```

Comment:
As stated in other comment, I don't think you need this parameter.
If you do decide to keep this parameter...
If the nullness of a parameter is going to control a critical thing like this, then it should be well document in the jsDoc for that parameter. It would not be obvious to a user of this class that setting a non-null value for anchorAdAttributes would cause an anchor to be placed even if isFilledAnchorAdEnabled returned false.

Reply:
Done.